lib/multipart.js

```diff
@@ -39,15 +39,20 @@
     function initiateUpload () {
       self.post(format('%s?uploads', filename), '', headers, options, function (err, res) {
         if (err) return ws.emit('error', err);
-        try {
-          var body = XML.parse(res.body).InitiateMultipartUploadResult;
+        parseXMLString(res.body, function (err, result) {
+          var body;
+          if (err) {
+            return ws.emit('error', err);
+          }
+          if (!result) {
```

Comment:
Put a space after `!` and remove brackets to be consistent with the rest of the code please (same above for `if (err)`.